src/test/java/com/bunq/sdk/BunqSdkTestBase.java

```diff
@@ -4,51 +4,190 @@
+    if (doesTestConfFileExist()) {
+      ApiContext apiContext = ApiContext.restore(TEST_CONFIG_PATH);
+      apiContext.ensureSessionActive();
+      apiContext.save(TEST_CONFIG_PATH);
+
+      return apiContext;
+    } else {
+      SandboxUser sandboxUser = generateNewSandboxUser();
+      ApiContext apiContext = ApiContext.create(
+          ApiEnvironmentType.SANDBOX,
+          sandboxUser.getApiKey(),
+          DEVICE_DESCRIPTION
+      );
+      apiContext.save(TEST_CONFIG_PATH);
+
+      return apiContext;
+    }
+  }
+
+  private static boolean doesTestConfFileExist() {
+    File confFile = new File(TEST_CONFIG_PATH);
+
+    return confFile.exists() && !confFile.isDirectory();
+  }
+
+  private static SandboxUser generateNewSandboxUser() {
+    OkHttpClient client = new OkHttpClient();
+
+    Request request = new Request.Builder()
+        .url(
+            SCHEME_HTTPS +
+            ApiEnvironmentType.SANDBOX.getBaseUri() +
+                URL_PATH_SEPARATOR +
+            ApiEnvironmentType.SANDBOX.getApiVersion() +
+            URL_PATH_SANDBOX_USER
+        )
+        .post(RequestBody.create(null, new byte[INDEX_FIRST]))
+        .addHeader(ApiClient.HEADER_REQUEST_ID, UUID.randomUUID().toString())
+        .addHeader(ApiClient.HEADER_CACHE_CONTROL, ApiClient.CACHE_CONTROL_NONE)
+        .addHeader(ApiClient.HEADER_GEOLOCATION, ApiClient.GEOLOCATION_ZERO)
+        .addHeader(ApiClient.HEADER_LANGUAGE, ApiClient.LANGUAGE_EN_US)
+        .addHeader(ApiClient.HEADER_REGION, ApiClient.REGION_NL_NL)
+        .build();
 
     try {
-      apiContext = ApiContext.restore(apiConfigPath);
-      apiContext.ensureSessionActive();
-    } catch (BunqException exception) {
-      List<String> ips = Arrays.asList(permittedIps);
-      apiContext = ApiContext.create(ApiEnvironmentType.SANDBOX, apiKey, DEVICE_DESCRIPTION, ips);
+      Response response = client.newCall(request).execute();
+      if (response.code() == HTTP_STATUS_OK) {
+        String responseString = response.body().string();
+        JsonObject jsonObject = new Gson().fromJson(responseString, JsonObject.class);
+        JsonObject apiKEy =jsonObject.getAsJsonArray(FIELD_RESPONSE)
+            .get(INDEX_FIRST)
+            .getAsJsonObject()
+            .get(FIELD_API_KEY)
+            .getAsJsonObject();
+
+        return SandboxUser.fromJsonReader(new JsonReader(new StringReader(apiKEy.toString())));
+      } else {
+        throw new BunqException(String.format(ERROR_COULD_NOT_GENERATE_NEW_API_KEY, response.body().string()));
+      }
+    } catch (IOException e) {
+      throw new BunqException(e.getMessage());
     }
+  }
+
+  private static void setSecondMonetaryAccountBank() {
+    BunqResponse<Integer> response = MonetaryAccountBank.create(CURRENCY_EUR, ACCOUNT_DESCRIPTION);
 
-    apiContext.save(apiConfigPath);
+    secondMonetaryAccountBank = MonetaryAccountBank.get(response.getValue()).getValue();
+  }
+
+  private static void requestSpendingMoney() {
```

Comment:
People using the SDK might not really understand what's actually happening here, maybe just add a comment.

Reply:
but, this is internal lib ? User shouldn't even know whats is happening here ? 🤔

Reply:
It's open source. People might want to look into why a request was randomly created and paid.

Reply:
in the tests ? :P

Reply:
In the test. We actually mention how to run the tests in the README.md. 

In this case the method name is clear but the process is quite weird (the sugar daddy stuff), I'm also not a big fan of comments but somethimes stuff happens outside of the code and it's not clear that's where you want to explain a bit.

This is open source code so people will actually look at it to understand, it's not something internal where we just point to a wiki. Or you can just create a wiki for the github page :P

Reply:
Will write a comment in then 👍